specs/core/1_shard-data-chains.md

```diff
@@ -230,20 +250,16 @@ def get_shard_header(block: ShardBlock) -> ShardBlockHeader:
 
 ```python
 def verify_shard_attestation_signature(state: BeaconState,
-                                       attestation: ShardAttestation) -> None:
-    data = attestation.data
-    persistent_committee = get_persistent_committee(state, data.shard, data.slot)
-    pubkeys = []
-    for i, index in enumerate(persistent_committee):
-        if attestation.aggregation_bits[i]:
-            validator = state.validators[index]
-            assert is_active_validator(validator, get_current_epoch(state))
-            pubkeys.append(validator.pubkey)
-    assert bls_verify(
-        pubkey=bls_aggregate_pubkeys(pubkeys),
-        message_hash=data.shard_block_root,
-        signature=attestation.aggregate_signature,
-        domain=get_domain(state, DOMAIN_SHARD_ATTESTER, compute_epoch_of_slot(data.slot))
+                                       block: ShardBlock) -> bool:
+    attester_committee = get_shard_block_attester_committee(state, block.shard, block.slot)
+    attesters = [v for i, v in enumerate(attester_committee) if i is not None and block.attester_bitfield[i]]
+    if get_shard_block_proposer_index(state, block.shard, block.slot) not in attesters:
```

Comment:
I think the logic is off here. The attesters included in block at `N` would be those that attested to block at `N-1`. Only the proposer knows this block to sign when it's created. There is also currently a circularity issue with the signature being of `block` but the block including who signed it in the bitfield.

I think we need two signatures -- `block.signature` (proposer signature) and `block.attester_signature` (aggregate attestation signature of slot `block.slot - 1`

Reply:
Ah, I'm doing things differently here. The attestations included in block N are attestations _of_ block N. So a block would be broadcasted with just one signature (the proposer) and others would add their signatures in.

I'm happy to go back to the other way, though it is technically ~96 bytes and 1 pairing less efficient :)